pkg/reconciler/clusterimagepolicy/controller.go

```diff
@@ -47,19 +48,19 @@ func NewController(
 	cmw configmap.Watcher,
 ) *controller.Impl {
 	clusterimagepolicyInformer := clusterimagepolicyinformer.Get(ctx)
-	nsSecretInformer := nsinformerfactory.Get(ctx).Core().V1().Secrets()
-	nsConfigMapInformer := nsinformerfactory.Get(ctx).Core().V1().ConfigMaps()
+	secretInformer := secretinformer.Get(ctx)
+	configMapInformer := cminformer.Get(ctx)
 
 	// Start the informers we got from the SharedInformerFactory above because
 	// injection doesn't do that for us since we're injecting the Factory and
 	// not the informers.
-	if err := controller.StartInformers(ctx.Done(), nsSecretInformer.Informer(), nsConfigMapInformer.Informer()); err != nil {
+	if err := controller.StartInformers(ctx.Done(), secretInformer.Informer(), configMapInformer.Informer()); err != nil {
```

Comment:
I know this is just a draft, but jotting this down.
Now that both of these are coming from the informers, instead of the factory, I think we do not need to manually start them.
https://github.com/knative/pkg/blob/main/injection/clients/namespacedkube/informers/core/v1/configmap/configmap.go#L36

For example, above, registers cm informer, so it should be started automagically by the sharedwithmain.

Reply:
Done